packages/core/content-releases/server/src/services/validation.ts

```diff
@@ -69,13 +69,17 @@ const createReleaseValidationService = ({ strapi }: { strapi: LoadedStrapi }) =>
       throw new errors.ValidationError('You have reached the maximum number of pending releases');
     }
   },
-  async validateUniqueNameForPendingRelease(name: CreateRelease.Request['body']['name']) {
+  async validateUniqueNameForPendingRelease(
+    name: CreateRelease.Request['body']['name'],
+    id?: UpdateRelease.Request['params']['id']
+  ) {
     const pendingReleases = (await strapi.entityService.findMany(RELEASE_MODEL_UID, {
       filters: {
         releasedAt: {
           $null: true,
         },
         name,
+        ...(id && { id: { $ne: id } }),
```

Comment:
As a general note! I would be careful with the &&  shortcircuiting,

It is definitely NOT an issue at all in this case, but I am writting the comment because I have seen this behaviour in some cases in strapi:

```ts
// Someone hijacking the id and making it be an object
const id = { id: {$ne: null} }

// Creating the filter
{
  //...
  ...{ id && { id: { $ne: id } })  // resolves to  { id: {$ne: null} }
}
```

Reply:
Really interesting 🤔 

I'm always careful with this kind of "sql injections" but didn't considered this one.

What do you think is the right way to do this? And do you think we should change it here?

Reply:
I believe this particular case it's not an issue, some options to avoid:

```ts
{
  //...
  ...{ !!id && { id: { $ne: id } })  // resolves to {}
}
```

other options:
```ts
// Someone hijacking the id and making it be an object
const id = { name: {$ne: null} }

// Creating the filter
{
  //...
  ...{ id ? { id: { $ne: id } }: {}})  // resolves to { } 
}
```